pkg/autoscaler/autoscaler/calculate/storage.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package calculate
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	promClient "github.com/prometheus/client_golang/api"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+func CalculateWhetherStoragePressure(tac *v1alpha1.TidbClusterAutoScaler, capacitySq, availableSq *SingleQuery,
+	client promClient.Client, metric v1alpha1.CustomMetric) (bool, error) {
+	if metric.Resource == nil ||
+		metric.Resource.Name != corev1.ResourceStorage ||
+		metric.LeastRemainAvailableStoragePercent == nil {
+		return false, fmt.Errorf("tac[%s/%s] didn't set storage metric correctly", tac.Namespace, tac.Name)
+	}
+
+	// query total available storage size
+	resp := &Response{}
+	err := queryMetricsFromPrometheus(tac, client, availableSq, resp)
+	if err != nil {
+		return false, err
```

Comment:
Suggest constructing a more clear error here, so that we do not have to add again like we do in #2851 #2885

Reply:
Please also fix the other similar cases.

Reply:
updated.